include/spdlog/sinks/eventlog_sink_win32.h

```diff
@@ -0,0 +1,246 @@
+// Copyright(c) 2020 bandana2004@gmail.com
+// Distributed under the MIT License (http://opensource.org/licenses/MIT)
+
+namespace eventlog_sink_win32 {
+
+struct win32_error : public spdlog_ex
+{
+    static std::string format(std::string const& func, DWORD error = GetLastError())
+    {
+        return fmt::format(func + " failed ({0})", error);
+    }
+
+    static void report(char const* message) SPDLOG_NOEXCEPT
+    {
+        fprintf(stderr, "%s", message);
+        fflush(stderr);
+    }
+
+    static void report(std::string const& message) SPDLOG_NOEXCEPT
+    {
+        report(message.c_str());
+    }
+
+    win32_error(std::string const& func_name, DWORD error = GetLastError())
+        : spdlog_ex(format(func_name, error), error)
+    {}
+};
+
+struct sid
+{
+    static SID* duplicate(SID* p_src_sid)
```

Comment:
Please add some comments It is really hard to understand what is the purpose of those function